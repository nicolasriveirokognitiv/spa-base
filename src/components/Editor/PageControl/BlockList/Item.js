import React from 'react';
import PropTypes from 'prop-types';
import getClassName from 'Utils/getClassName';
import { blockPropTypes } from 'Constants/propTypes';
import useTestMode from 'Hooks/useTestMode';

function Item({ onClick, isOpen, block, className }) {
    const { name, icon, type, id } = block.properties;

    const testProps = useTestMode(`item-${type}-${id}`);
    const itemClassName = getClassName({ isOpen }, 'page-control-item', className);

    return (
        <div className={itemClassName} onClick={() => !isOpen && onClick(isOpen, block)} {...testProps}>
            <div className="page-control-item-header">
                <span className={`page-control-item-header-icon bm-icon-${icon}`} />
                <span className="page-control-item-header-name">{name}</span>
            </div>
            {isOpen && (
                <div className="page-control-item-close" onClick={() => onClick(block)}>
                    <span className="bm-icon-times" />
                </div>
            )}
        </div>
    );
}

Item.defaultProps = {
    className : '',
    isOpen    : false,
};

Item.propTypes = {
    className : PropTypes.string,
    isOpen    : PropTypes.bool,
    onClick   : PropTypes.func.isRequired,
    block     : blockPropTypes.isRequired,
};

export default Item;
