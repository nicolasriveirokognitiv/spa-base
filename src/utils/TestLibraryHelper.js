/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
