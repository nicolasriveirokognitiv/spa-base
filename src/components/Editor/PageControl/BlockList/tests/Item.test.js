import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import Item from '../Item';

describe('Item', () => {
    let props;
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            onClick       : jest.fn(),
            itemClassName : jest.fn(),
            isOpen        : true,
            block         : {
                properties: {
                    name : 'foo',
                    icon : 'icon',
                },
            },
        };
    });

    test('Should render Correctly', () => {
        expect(<Item {...props} />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Item {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
