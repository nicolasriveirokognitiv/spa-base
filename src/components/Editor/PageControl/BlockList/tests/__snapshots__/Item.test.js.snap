// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-control-item  is-open"
  >
    <div
      class="page-control-item-header"
    >
      <span
        class="page-control-item-header-icon bm-icon-icon"
      />
      <span
        class="page-control-item-header-name"
      >
        foo
      </span>
    </div>
    <div
      class="page-control-item-close"
    >
      <span
        class="bm-icon-times"
      />
    </div>
  </div>
</DocumentFragment>
`;
